tests/cell/cell_test_suite.cpp

```diff
@@ -849,6 +853,328 @@ class cell_test_suite : public test_suite
         xlnt_assert_equals(cell2, cell3);
     }
 
+    void test_copy_value_between_workbooks()
+    {
+        // Test copying shared_string between different workbooks
+        xlnt::workbook wb_a;
+        auto ws_a = wb_a.active_sheet();
+        auto cell_a = ws_a.cell("A1");
+        cell_a.value("Test String");
+        xlnt_assert(cell_a.data_type() == xlnt::cell::type::shared_string);
+
+        xlnt::workbook wb_b;
+        auto ws_b = wb_b.active_sheet();
+        auto cell_b = ws_b.cell("B1");
+
+        // Copy value from workbook A to workbook B
+        cell_b.value(cell_a);
+
+        // Verify string is preserved correctly
+        xlnt_assert_equals(cell_b.value<std::string>(), "Test String");
+        xlnt_assert(cell_b.data_type() == xlnt::cell::type::shared_string);
+
+        // Verify workbooks are different
+        auto workbook_a = cell_a.workbook();
+        auto workbook_b = cell_b.workbook();
+        xlnt_assert(&workbook_a != &workbook_b);
```

Comment:
I think this doesn't really test anything. As it compare the addresses of different local variables (not references). Those will be always different. I guess, you actually want to test `xlnt_assert(workbook_a != workbook_b);`; which tests whether they contain different internal pointers.